Remove the management namespace (#183)
Resonate-Protocol/spec: pairing.md

2. **Dynamic Pairing Code** - pairing with a per-session [Pairing Code](README.md#definitions) that the client derives from a commit-and-reveal binding to the Noise handshake and emits via an out-channel (display, speaker, etc.) for the operator to enter into the server. See [Dynamic Pairing Code Flow](#dynamic-pairing-code-flow).
3. **Static Pairing Code** - pairing with a fixed [Pairing Code](README.md#definitions). Appropriate for devices with no out-channel; vulnerable to MITM if the pairing code is disclosed. See [Static Pairing Code Flow](#static-pairing-code-flow).

A code-based pairing starts from one of two states:

- **Unpaired.** The handshake used the Sentinel PSK; the channel is unauthenticated until the [PAKE](#pake) round completes. The round establishes trust from scratch and produces a new [long-term PSK](README.md#definitions).

> - **Already paired.** The server moves the established connection into pairing (see [Entering and leaving pairing](#entering-and-leaving-pairing)) and runs the round over the existing long-term PSK.

Reviewer: Question: Should we still keep this already paired case?

Reviewer 2: I'd say yes, arguably even more so now, as it becomes the only way to rotate the credential by the server.

Reviewer 3: I think that we can remove it. To rotate a credential a server can unpair the existing connection and then initiate pairing again. 

What would be the reason a server would want to rotate their keys? This is a feature that we can consider adding back later if there is demand.

The client reveals the new long-term PSK only after `server_kc` verifies, and only as `wrapped_psk` [sealed under the CPace output](#wrapping): a peer that cannot complete the PAKE - wrong pairing code, or a man in the middle relaying between two handshakes, whose differing `h` gives each leg a different `sid` - neither triggers the reveal nor can unwrap it.

Static pairing methods (Pairing PSK, Static Pairing Code) do not take over the device's out-channel. Dynamic pairing (Dynamic Pairing Code) takes over the out-channel - typically the audio output or display - to emit the per-session pairing code, so it cannot run while audio is playing on the same device; the operator must stop playback before initiating pairing (see [Multiple servers](connection.md#multiple-servers-server-initiated)).

Clients with a usable out-channel (display, speaker, etc.) SHOULD implement `dynamic_pairing_code` and prefer it to `static_pairing_code`, which is intended for devices without one. Clients whose display can render a QR code SHOULD also offer the `qr_code` [emission format](#dynamic-pairing-code-flow).

### Pairing Records

> Each successful pairing produces a pairing record: the new [long-term PSK](README.md#definitions) persisted together with the server's `server_id`. A repeat pairing with a server that already holds a record replaces that record.

Reviewer: ```suggestion
Each successful pairing produces a pairing record: the new [long-term PSK](README.md#definitions) persisted together with the server's `server_id`. The client MUST persist the new record, replacing any record it already holds for the server.
```

> A client MUST be able to store at least 5 pairing records; more is allowed. When a pairing completes at capacity, the client MUST evict an existing record so that the new record persists - a pairing never fails for lack of record storage. Which record is evicted is implementation-defined (for example, the least recently used), except that the client MUST NOT evict the record backing a currently-open connection.

Reviewer: @kahrendt What do you think about the 5 pairing records limit for ESPHome?
I think one of the reasons for the shared pairing records approach was to minimize storage usage on constraint devices.
This will require clients to reserve at least 320 bytes for Sendspin.

> A client MUST be able to store at least 5 pairing records; more is allowed. When a pairing completes at capacity, the client MUST evict an existing record so that the new record persists - a pairing never fails for lack of record storage. Which record is evicted is implementation-defined (for example, the least recently used), except that the client MUST NOT evict the record backing a currently-open connection.

Reviewer: Provisional connections can also be fully authenticated, was it ment to include those in the eviction too? If yes this should close the edge case:
```suggestion
A client MUST be able to store at least 5 pairing records; more is allowed. When a pairing completes at capacity, the client MUST evict an existing record so that the new record persists - a pairing never fails for lack of record storage. Which record is evicted is implementation-defined (for example, the least recently used), except that the client MUST NOT evict a record backing a currently-open connection, provisional or admitted; the client MUST cap its concurrently open paired connections below its record capacity (see [Multiple servers](connection.md#multiple-servers-server-initiated)) so an evictable record always exists.
```

### Unpaired Access

> A client MAY admit a server with no pairing record to activate roles or declare the `'playback'` activity. The session is [unpaired](README.md#definitions). Whether a client admits unpaired access is governed by its `unpaired_access` setting: the default is the manufacturer's choice, changing it is a local client action (manufacturer-defined), and the current value is advertised in [`client/hello`](messaging.md#client--server-clienthello) as `unpaired_access.enabled`.

Reviewer: Hmm, I think if a client disables this though a manufacturer defined local action during the playback activity this will cause undefined behavior.

Reviewer 3: Should we add that client sends `client/goodbye` with `pairing_required` ?

The Noise handshake completes using the pairing PSK, authenticating both sides. The client proceeds straight to [`client/pair-finalize`](#client--server-clientpair-finalize).

> **Lifecycle.** The client's pairing PSK MUST be drawn from a [CSPRNG](README.md#definitions) per device and MUST NOT be a fixed default shared across devices, whether provisioned at manufacture or generated by the client. It persists across reboots and is per-client and long-lived: a successful pairing does not consume it (pairing produces a separate [long-term PSK](README.md#definitions)), so it can pair the client with any number of servers.

Reviewer: Dropping the rotation prose also dropped "MUST NOT rotate it on its own", so nothing now forbids a client rotating this spontaneously. Just double checking, is that intentional to keep it completely up to the manufacturer?

Reviewer 3: Yep. We do not want to mandate how the client handles this.

Reviewer 3: Correct. For a client that is similar to dropping a pairing record, as the server won't be able to pair. I don't think it's worth mentioning. Just like a client can't start rewriting server_id, or mutate other data.

Decoding reverses the transform and MUST be lenient with operator-supplied input:

1. Trim surrounding whitespace and upper-case it.

> 2. If present, strip a leading `SP:`. The first character is the `version`; reject an unrecognized version.

Reviewer: This dropped the only place saying that an interface expecting a specific version rejects the others.
Now scanning the printed version 0 token into a `qr_code` prompt is undefined.

Pairing Window mechanics:

> - **Opening the window.** An operator gesture on the client - a physical button press, a reset-pinhole press, a button combo, a specific power-cycle pattern, a shake or motion gesture, or any equivalent implementation-defined action. Gestures SHOULD be deliberate and hard to induce remotely.

Reviewer: I think we should now require a client implementing `dynamic_pairing_code` to have one. Otherwise it can get locked out permanently (other than a factory reset) since we don't even have `management/open-pairing-window` now.

Reviewer 3: You can always turn a device off and on again.
